sponsorship_compassion/models/contracts.py

```diff
@@ -220,8 +220,12 @@ def write(self, vals):
         if 'child_id' in vals:
             self._on_change_child_id(vals)
 
+        if 'partner_id' in vals:
+            old_partner = self.partner_id

```

Comment:
By doing this you prevent the write method to be called on multiple records at the time. You should not do `self.partner_id` but you should check for all possible records in self (add a loop maybe)